### ISSUE TITLE
Error when response is compressed (gzip)

### ISSUE THREAD

**Issue by stealthv** [kolofordjango/kolo#10]
If the response has been compressed (for example using the `django.views.decorators.gzip.gzip_page` decorator), the following error is raised in `KoloProfiler.finalize_response`:

```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

Something like the following would fix that:
```python
import gzip

…

    def uncompressed_response_content(self, response: HttpResponse) -> str:
        if response.get("Content-Encoding") == "gzip":
            return gzip.decompress(response.content)
        return response.content

    def finalize_response(self, response: HttpResponse) -> None:
        duration = timezone.now() - self.timestamp

        self.response = {
            "ms_duration": round(duration.total_seconds() * 1000, 2),
            "status_code": response.status_code,
            "content": self.uncompressed_response_content(response).decode(
                response.charset
            ),
            "headers": dict(response.items()),
        }
        self.sql_queries_made = connection.queries

```

Not sure if there would be a need to support other compression types besides gzip

**Comment by wilhelmklopp** [comment]
👋 Hey @stealthv, thanks for opening this!

I have seen a similar error happen as a result of Kolo sitting above a compression middleware in the `MIDDLEWARE` setting. The solution there was to move the kolo middleware below the compression middleware (to ensure the compression happens after Kolo captures the response).

But I was unfamiliar with the approach of using a decorator for gziping, so thanks for sharing this error! We'll look into it

**Comment by Ian-Foote** [comment]
I've added a fix for this today. It should be available in the next release of Kolo.

**Comment by wilhelmklopp** [comment]
I've just published Kolo 1.3.1 which includes this fix 🎉 

https://pypi.org/project/kolo/

**Comment by abdulmuizzf** [comment]
@wilhelmklopp I got the same error for a request uploading a compressed zip file. 

`UnicodeDecodeError: 'utf-8' codec can't decode byte 0xea in position 165: invalid continuation byte`

Is it possible to fix this?

**Comment by wilhelmklopp** [comment]
@abdulmuizzf 👋 Thanks for the report. We just released version 1.3.7 of the kolo python package, which includes a fix for this bug 🙌 

Let me know if it's still not working :)
